posthog-core/src/index.ts

```diff
@@ -912,57 +912,41 @@ export abstract class PostHogCoreStateless {
     }
   }
 
-  /**
-   * Helper for flushing the queue in the background
-   * Avoids unnecessary promise errors
-   */
-  private flushBackground(): void {
-    void this.flush().catch(async (err) => {
-      await logFlushError(err)
-    })
-  }
-
-  async flush(): Promise<any[]> {
-    if (!this.flushPromise) {
-      this.flushPromise = this._flush().finally(() => {
-        this.flushPromise = null
-      })
-      this.addPendingPromise(this.flushPromise)
-    }
-    return this.flushPromise
-  }
-
-  protected getCustomHeaders(): { [key: string]: string } {
-    // Don't set the user agent if we're not on a browser. The latest spec allows
-    // the User-Agent header (see https://fetch.spec.whatwg.org/#terminology-headers
-    // and https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/setRequestHeader),
-    // but browsers such as Chrome and Safari have not caught up.
-    const customUserAgent = this.getCustomUserAgent()
-    const headers: { [key: string]: string } = {}
-    if (customUserAgent && customUserAgent !== '') {
-      headers['User-Agent'] = customUserAgent
-    }
-    return headers
-  }
-
-  private async _flush(): Promise<any[]> {
+  async flush(): Promise<PostHogQueueItem['message'][]> {
```

Comment:
```suggestion
  async flush(): Promise<void> {
```

it's a breaking change anyways and we don't use the return obj anywhere